### PR TITLE
add scalar load support for advanced path

### PR DESCRIPTION
**Pull request by Dewei-Wang-sh**

### CODE REVIEW

File: third_party/intel/lib/TritonIntelGPUToLLVM/TritonOpsToLLVM.cpp [hunk 1]
```diff
@@ -154,6 +154,18 @@ class LoadStorePrefetchOpConversion
...
     auto ptrType = cast<PointerType>(op.getPtr().getType());
+    // scalar load/store
+    if (!isa<RankedTensorType>(ptrType.getPointeeType())) {
+      if constexpr (std::is_same_v<OpType, LoadOp>) {
+        auto newLoad = rewriter.create<LLVM::LoadOp>(op.getLoc(), op.getType(),
+                                                     adaptor.getPtr());
+        rewriter.replaceOp(op, newLoad);
+        return success();
+      }
+      assert(0 && "add more support");
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/TritonOpsToLLVM.cpp:165]: can be replaced with `llvm_unreachable`

**Review by whitneywhtsang — COMMENTED** [review]
Do you think is worth adding a lit test or modifying an existing lit test to cover this case?

File: third_party/intel/lib/TritonIntelGPUToLLVM/TritonOpsToLLVM.cpp (hunk 1 above)
Comment (Dewei-Wang-sh) [reply third_party/intel/lib/TritonIntelGPUToLLVM/TritonOpsToLLVM.cpp:165]: sure, it's minor

**Comment by Dewei-Wang-sh** [comment]
> Do you think is worth adding a lit test or modifying an existing lit test to cover this case?

too trivial to add a lit IMO

**Pull request merged**